rego-java/src/main/java/com/contentgrid/opa/rego/ast/Term.java

```diff
@@ -99,7 +99,7 @@ public <T> T accept(RegoVisitor<T> visitor) {
     @AllArgsConstructor
     @NoArgsConstructor
     @EqualsAndHashCode(callSuper = true)
-    public static class ArrayTerm extends Term {
+    public static class ArrayTerm extends ScalarTerm<List<ScalarTerm<?>>> {
```

Comment:
Instead of having `ArrayTerm` extend `ScalarTerm`, which is technically not correct, because it is not a scalar term,
wouldn't it be more correct that the value is `List<Term>` instead?

Even though we are not looking to support it fully in thunx right now, technically an array can contain all other terms (at least according to [the rego grammar](https://www.openpolicyagent.org/docs/latest/policy-reference/#grammar))
`array           = "[" term { "," term } "]"`

Reply:
The only different between Term and Scalar term is that ScalarTerm has method `public abstract T getValue();`. As Collection can have value and I need to get it, that's why I extended ScalarTerm. Do you want me to move getValue() to Term?

Reply:
> Instead of having `ArrayTerm` extend `ScalarTerm`, which is technically not correct, because it is not a scalar term, wouldn't it be more correct that the value is `List<Term>` instead?
> 
> Even though we are not looking to support it fully in thunx right now, technically an array can contain all other terms (at least according to [the rego grammar](https://www.openpolicyagent.org/docs/latest/policy-reference/#grammar)) `array = "[" term { "," term } "]"`

Sorry for pinging, but could you please take a look on my comment bellow?

Reply:
Hi, sorry for the late response.

I think `ArrayTerm` should not implement `ScalarTerm`.

You can create a separate getter on `ArrayTerm` to retrieve the `List<Term>`, without having `ArrayTerm` implement `ScalarTerm`. 

Since instances of these classes are processed using a visitor-pattern anyways, it is not a problem that that `ArrayTerm` can not be cast to `ScalarTerm`. You get the concrete types in the visitor anyways, so the correct method can be called without needing any casting.